visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/Utils/EntityDtoUtil.java

```diff
@@ -2,30 +2,61 @@
 
 
 import com.petclinic.visits.visitsservicenew.DataLayer.Visit;
+import com.petclinic.visits.visitsservicenew.DomainClientLayer.PetResponseDTO;
+import com.petclinic.visits.visitsservicenew.DomainClientLayer.PetsClient;
+import com.petclinic.visits.visitsservicenew.DomainClientLayer.VetDTO;
+import com.petclinic.visits.visitsservicenew.DomainClientLayer.VetsClient;
 import com.petclinic.visits.visitsservicenew.PresentationLayer.VisitRequestDTO;
 import com.petclinic.visits.visitsservicenew.PresentationLayer.VisitResponseDTO;
+import lombok.RequiredArgsConstructor;
 import org.springframework.beans.BeanUtils;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
 
 import java.util.UUID;
 
-
+@Component
+@RequiredArgsConstructor
 public class EntityDtoUtil {
 
-    public static VisitResponseDTO toVisitResponseDTO(Visit visit) {
+    private final VetsClient vetsClient;
+    private final PetsClient petsClient;
+
+    public Mono<VisitResponseDTO> toVisitResponseDTO(Visit visit) {
         System.out.println("Entity Date in Mapping: " + visit.getVisitDate()); // Debugging
-        VisitResponseDTO visitResponseDTO = new VisitResponseDTO();
-        BeanUtils.copyProperties(visit, visitResponseDTO);
-        return visitResponseDTO;
+
+        Mono<PetResponseDTO> petResponseDTOMono = petsClient.getPetById(visit.getPetId());
+        Mono<VetDTO> vetResponseDTOMono = vetsClient.getVetByVetId(visit.getPractitionerId());
+
+        return Mono.zip(petResponseDTOMono, vetResponseDTOMono)
+                .flatMap(tuple -> {
+                    PetResponseDTO petResponseDTO = tuple.getT1();
+                    VetDTO vetResponseDTO = tuple.getT2();
```

Comment:
Can you clarify this?

Reply:
Mono.zip produces a tuple which can vary from type. In this case, there is Mono<PetResponseDTO> as the first item and Mono<VetDTO> as the second. I am then unzipping those items and passing the information that is inside of them to items of their respective types.